gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/anonymous/AnonymousConfig.groovy

```diff
@@ -31,18 +34,28 @@ class AnonymousConfig extends WebSecurityConfigurerAdapter {
   static String key = "spinnaker-anonymous"
   static String defaultEmail = "anonymous"
 
+  @Autowired
+  CredentialsService credentialsService
+
+  List<String> anonymousAllowedAccounts = []
+
   void configure(HttpSecurity http) {
-    def principal = new User(
-        email: defaultEmail,
-        roles: ["anonymous"],
-        allowedAccounts: []
-    )
+    // Not using the ImmutableUser version in order to update allowedAccounts.
+    def principal = new User(email: defaultEmail, allowedAccounts: anonymousAllowedAccounts)
 
-    http.anonymous()
+    http
+      .anonymous()
         .key(key)
-        .authorities("anonymous")
         .principal(principal)
         .and()
-        .csrf().disable()
+      .csrf()
+        .disable()
+  }
+
+  @Scheduled(initialDelay = 30000L, fixedDelay = 60000L)
```

Comment:
Done. Also removed initial delay - it isn't necessary if clouddriver is already up.